site/usage/typescript.md

```diff
@@ -0,0 +1,46 @@
+---
+layout: usage
+menu: usage
+title: Vega-Lite and Typescript
+permalink: /usage/typescript.html
+---
+
+Vega-Lite is written in [TypeScript](https://www.typescriptlang.org). You can use the types provided by Vega-Lite to type check your specifications and the compiler API.
+
+The main function for Vega-Lite is [`compile`](compile.html). Vega-Lite directly exports this function. The first argument to `compile` is a [Vega-Lite specification]({{ site.baseurl }}/docs/spec.html) of type `TopLevelSpec`. You can use the type, which Vega-Lite directly exports, to type check your specification. Vega-Lite also exports a type `Config` for the [Vega-Lite configuration]({{ site.baseurl }}/docs/config.html).
```

Comment:
Re: the issue in #7126, it would be helpful to note that using wrappers on vega-lite like vega-embed need to have their types constrained to avoid inference errors. Maybe that documentation should go on vega-embed instead?

Reply:
I don't think that's the case. 

```ts
import {compile} from 'vega-lite';

const vegaLiteSpec = {
  $schema: 'https://vega.github.io/schema/vega-lite/v4.json',
  data: {
    values: [
      {a: 'A', b: 28},
      {a: 'B', b: 55},
      {a: 'C', b: 43},
      {a: 'D', b: 91},
      {a: 'E', b: 81},
      {a: 'F', b: 53},
      {a: 'G', b: 19},
      {a: 'H', b: 87},
      {a: 'I', b: 52}
    ]
  },
  mark: 'bar',
  encoding: {
    x: {field: 'a', type: 'nominal', axis: {labelAngle: 0}},
    y: {field: 'b', type: 'quantitative'}
  }
} as const;

const vegaSpec = compile(vegaLiteSpec).spec;
```

The issue without `as const` is that Typescript widens the type of `"bar"` to `string` rather than the literal `"bar"`.